scripts/modtools/extra-gamelog.lua

```diff
@@ -0,0 +1,189 @@
+        else
+            msg(dfhack.TranslateName(dfhack.units.getVisibleName(expedition_leader)).." became expedition leader.")
+        end
+    end
+
+    if mayor ~= old_mayor then
+        if mayor == nil then
+            msg("Mayor position is now vacant.")
+        else
+            msg(dfhack.TranslateName(dfhack.units.getVisibleName(mayor)).." became mayor.")
+        end
+    end
+    old_mayor = mayor
+    old_expedition_leader = expedition_leader
+end
+
+siege = false
+function log_siege()
+    local function cur_siege()
+        for _, unit in ipairs(df.global.world.units.active) do
+            if unit.flags1.active_invader then return true end
+        end
+        return false
+    end
+    local old_siege = siege
+    siege = cur_siege()
+    if siege ~= old_siege and siege then
+        msg("A vile force of darkness has arrived!")
+    elseif siege ~= old_siege and not siege then
+        msg("Siege was broken.")
+    end
+end
+
+
+local workshopTypes = {
+    [0]="Carpenters Workshop",
+    "Farmers Workshop",
+    "Masons Workshop",
+    "Craftsdwarfs Workshop",
+    "Jewelers Workshop",
+    "Metalsmiths Forge",
+    "Magma Forge",
+    "Bowyers Workshop",
+    "Mechanics Workshop",
+    "Siege Workshop",
+    "Butchers Workshop",
+    "Leatherworks Workshop",
+    "Tanners Workshop",
+    "Clothiers Workshop",
+    "Fishery",
+    "Still",
+    "Loom",
+    "Quern",
+    "Kennels",
+    "Kitchen",
+    "Ashery",
+    "Dyers Workshop",
+    "Millstone",
+    "Custom",
+    "Tool",
+    }
+
+local furnaceTypes = {
+    [0]="Wood Furnace",
+    "Smelter",
+    "Glass Furnace",
+    "Kiln",
+    "Magma Smelter",
+    "Magma Glass Furnace",
+    "Magma Kiln",
+    "Custom Furnace",
+    }
+
+buildStates = {}
+
+function log_buildings()
+    for _, building in ipairs(df.global.world.buildings.all) do
+        if getmetatable(building) == "building_workshopst" or getmetatable(building) == "building_furnacest" then
+            buildStates[building.id] = buildStates[building.id] or building.flags.exists
+            if buildStates[building.id] ~= building.flags.exists then
+                buildStates[building.id] = building.flags.exists
+                if building.flags.exists then
+                    if getmetatable(building) == "building_workshopst" then
+                        msg(workshopTypes[building.type].." was built.")
+                    elseif getmetatable(building) == "building_furnacest" then
+                        msg(furnaceTypes[building.type].." was built.")
+                    end
+                end
+            end
+        end
+    end
+end
+
+local function event_loop()
+    log_nobles()
+    log_siege()
+    log_buildings()
+    dfhack.timeout(50, 'ticks', event_loop)
+end
+
+local args = {...}
+if args[1] == 'disable' then
+    dfhack.onStateChange[_ENV] = nil
+elseif args[1] == 'enable' then
+    dfhack.onStateChange[_ENV] = log_on_load
+    event_loop()
```

Comment:
This will keep running after the script is disabled. (I usually have a global `enabled` variable that I set here and check before scheduling the timeout again.)